cli/command/container/stop.go

```diff
@@ -39,16 +40,45 @@ func NewStopCommand(dockerCli *command.DockerCli) *cobra.Command {
 
 func runStop(dockerCli *command.DockerCli, opts *stopOptions) error {
 	ctx := context.Background()
+	timeout := time.Duration(opts.time) * time.Second
 
+	var waitAll sync.WaitGroup
 	var errs []string
+	const maxParallel int32 = 50
+
+	// make sure result is printed in order
+	output := map[string]chan error{}
 	for _, container := range opts.containers {
-		timeout := time.Duration(opts.time) * time.Second
-		if err := dockerCli.Client().ContainerStop(ctx, container, &timeout); err != nil {
-			errs = append(errs, err.Error())
-		} else {
-			fmt.Fprintf(dockerCli.Out(), "%s\n", container)
+		output[container] = make(chan error, 1)
+	}
+	waitAll.Add(1)
+	go func() {
+		for _, container := range opts.containers {
+			if err := <-output[container]; err != nil {
+				errs = append(errs, err.Error())
+			} else {
+				fmt.Fprintf(dockerCli.Out(), "%s\n", container)
+			}
 		}
+		waitAll.Done()
+	}()
```

Comment:
I don't think it's most important here but I still think it's more correct to keep order. In both cases the final duration of the command is same, by keeping it in order we still never show any output slower than we do atm and we have no chance of breaking existing users.

For readability of this goroutine I'd put it around the other range loop instead.